### PR TITLE
Fix crash when using a writable FileStream to a file in non-existing folder

### PR DESCRIPTION
**Pull request by Broxzier**
Something that I ran in to long ago, but got stuck at the time on an annoying issue with Windows's `CreateDirectory` macro. The second commit fixes this.

This change makes sure the directory exists before `_wfopen` is called.

### CODE REVIEW

File: src/openrct2/core/FileStream.hpp [hunk 1]
```diff
@@ -81,6 +82,16 @@ namespace OpenRCT2
                     throw;
             }
 
+            // Make sure the directory exists before writing to a file inside it
+            if (_canWrite)
+            {
+                std::string directory = Path::GetDirectory(path);
+                if (!Path::DirectoryExists(directory))
+                {
+                    Path::CreateDirectory(directory);
```
Comment (tupaschoal) [inline src/openrct2/core/FileStream.hpp:91]: Shouldn't we check the return and throw in case this fails?

Comment (Broxzier) [reply src/openrct2/core/FileStream.hpp:91]: `Path::CreateDirectory` is void and doesn't throw from the looks of it. It itself does call a function returning bool, however. It's probably a good idea to validate the return value there and return false or throw on failure.

Comment (tupaschoal) [reply src/openrct2/core/FileStream.hpp:91]: I don't mind if this is left for another PR, as it will take changing multiple places.

**Review by tupaschoal — COMMENTED** [review]
Does this fix any open issue? Probably could use a changelog

**Comment by Broxzier** [comment]
> Does this fix any open issue? Probably could use a changelog

I highly doubt it. With the current codebase, all folders for where FileStream gets currently used are created when launching the game. With this change, any future uses of FileStream (to files in folders that don't exist yet) would be working without crashing.

**Comment by tupaschoal** [comment]
Okz then I guess it's fine to merge without changelog

**Pull request merged**